internal/sqlsmith: TestGenerateParse failed

internal/sqlsmith.TestGenerateParse [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15849885?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15849885?buildTab=artifacts#/) on master @ [a14e17082a440a12a3c1019181de11e0a576068c](https://github.com/cockroachdb/cockroach/commits/a14e17082a440a12a3c1019181de11e0a576068c):


```
	c̄ol_88 DESC,
	col_89 DESC NULLS FIRST;

STMT: 19
SELECT
	*
FROM
	(
		SELECT
			tab_48."col1'_3" AS "c\\x59ol_90", tab_48.col1_1 AS "c""
ol_91", B'0000'::BIT(38) AS col_92
		FROM
			defaultdb.public."ta	ble_1" AS tab_48
		WHERE
			((tab_48."c/o\\x32l1_2"::STRING % tab_48.col1_1::STRING)::BOOL OR true::BOOL)
	)
		AS "t%baab _49"
ORDER BY
	"c\\x59ol_90" ASC NULLS FIRST, "c""
ol_91" ASC, col_92 DESC NULLS FIRST;

    setup_test.go:121: SELECT
        	*
        FROM
        	(
        		SELECT
        			tab_50."col1'_3" AS "c̰ol%v_93",
        			tab_50."c ol1_4" AS "coͭl%
        _94",
        			NULL::INT2VECTOR AS col_95,
        			tab_50.col1_1 AS co😗l_96,
        			tab_50.col1_1 AS "'col_97",
        			\x47::BYTES::BYTES AS col_98
        		FROM
        			defaultdb.public."ta	ble_1" AS tab_50
        		WHERE
        			false::BOOL
        	)
        		AS "T'a😥b_51"
        ORDER BY
        	"c̰ol%v_93" ASC NULLS LAST,
        	"coͭl%
        _94" NULLS LAST,
        	col_95 ASC NULLS FIRST,
        	co😗l_96 NULLS LAST,
        	"'col_97" DESC NULLS LAST,
        	col_98 ASC NULLS LAST: at or near "\": syntax error
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/f82616d5c9a8c3209014f997587365ae/logTestGenerateParse2242199906
--- FAIL: TestGenerateParse (1.69s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestGenerateParse.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

internal/sqlsmith.TestGenerateParse [failed](https://tanzanite.cluster.engflow.com/invocations/default/d0896a88-86c0-4def-a152-96f9e33b5dd2?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvaW50ZXJuYWwvc3Fsc21pdGg6c3Fsc21pdGhfdGVzdA==) on master @ [b4402507fdebe5682f77e348331a308d378c91e1](https://github.com/cockroachdb/cockroach/commits/b4402507fdebe5682f77e348331a308d378c91e1):


```
        			tab_155."co	l_222" AS col_224,
        			'!''nWTTQA'''::TSQUERY::TSQUERY AS col_225
        		FROM
        			(
        				VALUES
        					(NULL::PG_LSN),
        					(NULL::PG_LSN),
        					(pg_lsn('C'::STRING::STRING::STRING)::PG_LSN),
        					(NULL::PG_LSN),
        					(
        						CASE
        						WHEN (inet('::1/0'::INET::INET::INET)::INET::INET >> '::8/0'::INET::INET::INET)::BOOL
        						THEN NULL::PG_LSN
        						ELSE NULL::PG_LSN
        						END
        					),
        					(NULL::PG_LSN)
        			)
        				AS tab😢_152 (col_218),
        			(
        				VALUES
        					('!''QZYtAwQCr'''::TSQUERY::TSQUERY),
        					('''rP'' & !''ZXlZ'' <-> ''kX'''::TSQUERY::TSQUERY),
        					('''P'' <-> ''UdzNGWA'' | ''VyegpKSUcF'''::TSQUERY::TSQUERY),
        					(NULL::TSQUERY),
        					('''DutMB'''::TSQUERY::TSQUERY),
        					(NULL::TSQUERY)
        			)
        				AS tab_153 ("c
        ol
_219"),
        			(
        				VALUES
        					(\x45::BYTES::BYTES),
        					(
        						digest(\x47::BYTES::BYTES::BYTES, (SELECT NULL::STRING AS "col _221" FROM (VALUES (int4('{"D": 2, "F": {"A": null, "G": false}, "J": [[{"A": {"B": "D", "D": "D", "J": false}}, [{"F": [true]}], true], [2], 1]}'::JSONB::JSONB::JSONB)::INT4), (int4('2000-01-01 09:00:00'::TIMESTAMP::TIMESTAMP::TIMESTAMP)::INT4), (6::INT4), (7::INT4), (5::INT4)) AS "taB_154" ("""😈co l_220") ORDER BY "col _221" DESC NULLS LAST LIMIT 1)::STRING)::BYTES
        					),
        					(\x42::BYTES::BYTES),
        					(\x4a::BYTES::BYTES)
        			)
        				AS tab_155 ("co	l_222")
        		ORDER BY
        			tab😢_152.col_218 NULLS FIRST, tab_153."c
        ol
_219" ASC NULLS LAST, tab_155."co	l_222" DESC NULLS FIRST
        	)
        		AS tab_156
        ORDER BY
        	" Col_223" ASC NULLS LAST, col_224 DESC NULLS LAST, col_225 NULLS LAST: at or near "\": syntax error
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestGenerateParse80883509
--- FAIL: TestGenerateParse (1.37s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestGenerateParse.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

internal/sqlsmith.TestGenerateParse [failed](https://tanzanite.cluster.engflow.com/invocations/default/92aad65a-19af-4896-a3f3-7abac59dc80b?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvaW50ZXJuYWwvc3Fsc21pdGg6c3Fsc21pdGhfdGVzdA==) on master @ [b4402507fdebe5682f77e348331a308d378c91e1](https://github.com/cockroachdb/cockroach/commits/b4402507fdebe5682f77e348331a308d378c91e1):


```
    setup_test.go:121: SELECT
        	*
        FROM
        	(
        		SELECT
        			tab😏_79."co'l_95" AS col_96,
        			st_geogfromwkb(\x42::BYTES::BYTES::BYTES, 3::INT4::INT4)::GEOGRAPHY AS "c%vol_97",
        			"t	ab_70".col_85 AS col_98,
        			"Tͬ%qab_78".col_94 AS col_99,
        			"t%pab_69"."%qcol_84" AS col_100,
        			"t%pab_69".col_83 AS col_101
        		FROM
        			(
        				VALUES
        					(NULL::BIT(43), NULL::CHAR),
        					(B'0110'::BIT(43), 'F'::STRING::CHAR),
        					(B'0100'::BIT(43), 'A'::STRING::CHAR)
        			)
        				AS "t%pab_69" (col_83, "%qcol_84"),
        			(VALUES ('2000-01-10'::DATE::DATE), (NULL::DATE), ('2000-01-06'::DATE::DATE), ('2000-01-06'::DATE::DATE))
        				AS "t	ab_70" (col_85),
        			(VALUES (NULL::GEOGRAPHY), (NULL::GEOGRAPHY)) AS tab_73 (col_89)
        			JOIN (
        					VALUES
        						(8.0::FLOAT8),
        						(date_part('I'::STRING::STRING::STRING, '00:03:00'::TIME::TIME::TIME)::FLOAT8),
        						(8.0::FLOAT8),
        						(3.0::FLOAT8)
        				)
        					AS "Tͬ%qab_78" (col_94)
        				FULL JOIN (VALUES ('::3/0'::INET::INET), ('::8/0'::INET::INET)) AS tab😏_79 ("co'l_95") ON
        						tab😏_79."co'l_95"::INET
        						IN (tab😏_79."co'l_95", tab😏_79."co'l_95", tab😏_79."co'l_95") ON
        					(false::BOOL AND true::BOOL)
        		WHERE
        			true::BOOL
        		ORDER BY
        			"t%pab_69"."%qcol_84" ASC NULLS LAST
        	)
        		AS tab_80
        ORDER BY
        	col_96 NULLS FIRST,
        	"c%vol_97" DESC NULLS LAST,
        	col_98 ASC NULLS LAST,
        	col_99 DESC NULLS LAST,
        	col_100 DESC NULLS LAST,
        	col_101 ASC NULLS FIRST: at or near "\": syntax error
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestGenerateParse407789544
--- FAIL: TestGenerateParse (6.99s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestGenerateParse.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

internal/sqlsmith.TestGenerateParse [failed](https://tanzanite.cluster.engflow.com/invocations/default/a0a0b3b4-949f-4976-8f1f-bf0563438a1e?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvaW50ZXJuYWwvc3Fsc21pdGg6c3Fsc21pdGhfdGVzdA==) on master @ [b4402507fdebe5682f77e348331a308d378c91e1](https://github.com/cockroachdb/cockroach/commits/b4402507fdebe5682f77e348331a308d378c91e1):


```
        FROM
        	(
        		SELECT
        			'::3/0'::INET::INET AS col_80,
        			tab_53._enum AS col_81,
        			tab_53._float8 AS co🙄l_82,
        			\x49::BYTES::BYTES AS col_83,
        			NULL::DATE AS "c\\u52D4ol_84",
        			tab_53._int4 AS "co
l_85"
        		FROM
        			defaultdb.public.seed AS tab_53
        		ORDER BY
        			co🙄l_82 DESC NULLS FIRST,
        			tab_53._float4,
        			"co
l_85" ASC,
        			tab_53._timestamptz ASC NULLS FIRST,
        			tab_53._bool NULLS LAST,
        			tab_53._inet,
        			col_83 NULLS FIRST,
        			tab_53._timestamp,
        			"c\\u52D4ol_84" NULLS LAST,
        			tab_53._decimal DESC,
        			tab_53._date NULLS FIRST,
        			tab_53._int2 NULLS FIRST,
        			tab_53._interval ASC,
        			tab_53._bytes DESC NULLS FIRST,
        			tab_53._int8 NULLS LAST,
        			tab_53._int4 ASC NULLS LAST,
        			tab_53._enum,
        			col_80 ASC NULLS LAST,
        			tab_53.tableoid NULLS LAST,
        			tab_53._jsonb ASC,
        			tab_53._string DESC NULLS FIRST,
        			col_81 DESC NULLS LAST,
        			tab_53._float8 DESC NULLS LAST,
        			tab_53._uuid DESC NULLS FIRST
        		LIMIT
        			61
        	)
        		AS "😱ta͐b _54"
        ORDER BY
        	col_80,
        	col_81 DESC NULLS FIRST,
        	co🙄l_82 NULLS FIRST,
        	col_83 ASC NULLS LAST,
        	"c\\u52D4ol_84" NULLS FIRST,
        	"co
l_85" DESC NULLS LAST: at or near "\": syntax error
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestGenerateParse392340570
--- FAIL: TestGenerateParse (12.52s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestGenerateParse.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

internal/sqlsmith.TestGenerateParse [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15859842?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15859842?buildTab=artifacts#/) on master @ [fbb6f16be385b6238b5b8cbe5a5560716b6849a4](https://github.com/cockroachdb/cockroach/commits/fbb6f16be385b6238b5b8cbe5a5560716b6849a4):


```
	col😬_40 NULLS FIRST,
	""" col̉'_41" DESC NULLS FIRST,
	"coL_42" DESC NULLS LAST,
	col_43 DESC NULLS LAST,
	col_44 DESC NULLS FIRST,
	"co-l_45";

STMT: 6
SELECT
	*
FROM
	(
		SELECT
			2::DECIMAL AS "c)ol%p_46", ("T͗Ab_17"._bytes::BYTES || "ta(b_18"._bytes::BYTES)::BYTES AS col_47
		FROM
			defaultdb.public.seed AS tab_15,
			defaultdb.public.seed AS tab_16,
			defaultdb.public.seed AS "T͗Ab_17",
			defaultdb.public.seed AS "ta(b_18"
		WHERE
			(tab_16._inet::INET >> "T͗Ab_17"._inet::INET)::BOOL
	)
		AS "%vta😒b_19"
ORDER BY
	"c)ol%p_46" DESC, col_47 DESC NULLS FIRST;

    setup_test.go:121: SELECT
        	*
        FROM
        	(
        		SELECT
        			2.0::FLOAT8 AS col_48,
        			\x42::BYTES::BYTES AS col_49,
        			tab_20._decimal AS "cOl_50",
        			tab_20._bool AS col_51,
        			NULL::INET AS " col_52",
        			'2000-01-01 06:00:00'::TIMESTAMP::TIMESTAMP AS col😢_53
        		FROM
        			defaultdb.public.seed AS tab_20
        	)
        		AS tab_21
        ORDER BY
        	col_48 NULLS FIRST,
        	col_49 DESC NULLS FIRST,
        	"cOl_50" NULLS FIRST,
        	col_51 ASC NULLS LAST,
        	" col_52" ASC NULLS LAST,
        	col😢_53: at or near "\": syntax error
    panic.go:626: -- test log scope end --
--- FAIL: TestGenerateParse (2.24s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestGenerateParse.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

internal/sqlsmith.TestGenerateParse [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15861192?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15861192?buildTab=artifacts#/) on master @ [74d4d710873bb576b6ad9bf57c9d36358a4c8caf](https://github.com/cockroachdb/cockroach/commits/74d4d710873bb576b6ad9bf57c9d36358a4c8caf):


```
		WHERE
			NULL::BOOL
		ORDER BY
			tab_53.col_68, tab_53.col_68 DESC NULLS LAST, tab_53.col_68 NULLS LAST
	)
		AS "tab _54"
ORDER BY
	col😽_69 NULLS LAST;

STMT: 24
SELECT
	*
FROM
	(
		SELECT
			t😏ab_55."""col_70" AS col_71
		FROM
			(VALUES (NULL::GEOMETRY), (NULL::GEOMETRY), (NULL::GEOMETRY), (NULL::GEOMETRY), (NULL::GEOMETRY))
				AS t😏ab_55 ("""col_70")
	)
		AS tab_56
ORDER BY
	col_71 ASC NULLS FIRST;

    setup_test.go:121: SELECT
        	*
        FROM
        	(
        		SELECT
        			"
😑tab_57".col_72 AS col_73
        		FROM
        			(
        				VALUES
        					(\x44::BYTES::BYTES),
        					(NULL::BYTES),
        					(\x44::BYTES::BYTES),
        					(\x41::BYTES::BYTES),
        					(NULL::BYTES),
        					(\x49::BYTES::BYTES)
        			)
        				AS "
😑tab_57" (col_72)
        		WHERE
        			NULL::PG_LSN::PG_LSN NOT IN (NULL::PG_LSN,)
        	)
        		AS tab_58
        ORDER BY
        	col_73 DESC: at or near "\": syntax error
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/f82616d5c9a8c3209014f997587365ae/logTestGenerateParse3967430157
--- FAIL: TestGenerateParse (0.86s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestGenerateParse.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

internal/sqlsmith.TestGenerateParse [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15861654?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15861654?buildTab=artifacts#/) on master @ [13d8e47842d0e5e9f5887b4ad1cb821f0e467abc](https://github.com/cockroachdb/cockroach/commits/13d8e47842d0e5e9f5887b4ad1cb821f0e467abc):


```
	)
		AS " tab_26"
ORDER BY
	😯col_20 ASC NULLS LAST, " %f5Col_21" ASC NULLS FIRST;

    setup_test.go:121: SELECT
        	*
        FROM
        	(
        		SELECT
        			tab_27.col̶3_6 AS "c%e8ol_22",
        			tab_27.col3_0 AS co😞l_23,
        			(tab_27.col3_4 AND tab_27.col3_4) AS "co
        l'_24",
        			tab_27.col3_11 AS colͪ_25,
        			NULL::VOID AS "col%p🙃_26",
        			tab_27.col3_11 AS "c_o
l�_27",
        			tab_27.col3_2 AS col_28,
        			\x47::BYTES::BYTES AS col😦_29,
        			tab_27."co'%pl3_3" AS "cOl_30",
        			tab_27.col3_12 AS col_31,
        			tab_27.co😿l3_8 AS col_32,
        			tab_27.c😡ol3_5 AS col_33,
        			tab_27.col3_10 AS col_34,
        			tab_27."col\\x923_9" AS col_35
        		FROM
        			defaultdb.public."tabl
e_3" AS tab_27
        		ORDER BY
        			tab_27.col3_10 DESC
        	)
        		AS "%vtab😱_28"
        ORDER BY
        	"c%e8ol_22" NULLS FIRST,
        	co😞l_23 DESC NULLS LAST,
        	"co
        l'_24",
        	colͪ_25 DESC NULLS LAST,
        	"col%p🙃_26" NULLS LAST,
        	"c_o
l�_27" ASC NULLS LAST,
        	col_28 NULLS FIRST,
        	col😦_29 NULLS LAST,
        	"cOl_30" ASC NULLS LAST,
        	col_31 NULLS LAST,
        	col_32 NULLS FIRST,
        	col_33 ASC NULLS FIRST,
        	col_34 ASC NULLS FIRST,
        	col_35 DESC: at or near "\": syntax error
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/f82616d5c9a8c3209014f997587365ae/logTestGenerateParse3245938869
--- FAIL: TestGenerateParse (2.56s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestGenerateParse.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/internal/sqlsmith/BUILD.bazel
+++ b/pkg/internal/sqlsmith/BUILD.bazel
@@ -64,7 +64,6 @@ go_test(
         "//pkg/server",
         "//pkg/sql/parser",
         "//pkg/testutils/serverutils",
-        "//pkg/testutils/skip",
         "//pkg/testutils/sqlutils",
         "//pkg/testutils/testcluster",
         "//pkg/util/leaktest",

--- a/pkg/internal/sqlsmith/setup_test.go
+++ b/pkg/internal/sqlsmith/setup_test.go
@@ -21,7 +21,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/internal/sqlsmith"
 	"github.com/cockroachdb/cockroach/pkg/sql/parser"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
-	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
@@ -65,8 +64,6 @@ func TestSetups(t *testing.T) {
 func TestGenerateParse(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-
-	skip.WithIssue(t, 126344)
 
 	ctx := context.Background()
 	srv, sqlDB, _ := serverutils.StartServer(t, base.TestServerArgs{})

--- a/pkg/sql/logictest/testdata/logic_test/pg_catalog
+++ b/pkg/sql/logictest/testdata/logic_test/pg_catalog
@@ -5169,10 +5169,13 @@ subtest regression_126042
 statement ok
 CREATE TABLE t126042 (
   id INT8 NOT NULL PRIMARY KEY,
-  bytes_array BYTES[] DEFAULT '{f}'::BYTES[]
+  b BYTES DEFAULT 'f'::BYTES,
+  bytes_array BYTES[] DEFAULT '{f}'::BYTES[],
+  text_array TEXT[] DEFAULT '{a, b}'::TEXT[]
 )
 
-# Regression test for #126042. Expressions of type BYTES[] should be parsable.
+# Regression test for #126042 and #126344. Expressions of type BYTES and BYTES[]
+# should be parsable.
 query TT
 SELECT a.attname, pg_get_expr(d.adbin, d.adrelid)
 FROM pg_attribute a
@@ -5180,8 +5183,10 @@ LEFT JOIN pg_attrdef d ON a.attrelid = d.adrelid AND a.attnum = d.adnum
 WHERE a.attrelid = 't126042'::REGCLASS
 ORDER BY 1
 ----
+b            '\x66'::BYTES
 bytes_array  '{"\\x66"}'::BYTES[]
 id           NULL
+text_array   '{a,b}'::STRING[]
 
 subtest end
 

--- a/pkg/sql/sem/tree/datum.go
+++ b/pkg/sql/sem/tree/datum.go
@@ -1563,7 +1563,7 @@ func writeAsHexString(ctx *FmtCtx, b string) {
 // Format implements the NodeFormatter interface.
 func (d *DBytes) Format(ctx *FmtCtx) {
 	f := ctx.flags
-	if f.HasFlags(fmtPgwireFormat) || f.HasFlags(fmtPGCatalog) {
+	if f.HasFlags(fmtPgwireFormat) {
 		ctx.WriteString(`\x`)
 		writeAsHexString(ctx, string(*d))
 	} else if f.HasFlags(fmtFormatByteLiterals) {
@@ -1572,7 +1572,7 @@ func (d *DBytes) Format(ctx *FmtCtx) {
 		_, _ = hex.NewEncoder(ctx).Write([]byte(*d))
 		ctx.WriteByte('\'')
 	} else {
-		withQuotes := !f.HasFlags(FmtFlags(lexbase.EncBareStrings))
+		withQuotes := !f.HasFlags(FmtBareStrings)
 		if withQuotes {
 			ctx.WriteByte('\'')
 		}
@@ -4907,6 +4907,7 @@ func (d *DArray) Format(ctx *FmtCtx) {
 	if ctx.flags.HasAnyFlags(fmtPgwireFormat | fmtPGCatalog) {
 		defer func(f FmtFlags) { ctx.flags = f }(ctx.flags)
 		ctx.flags = ctx.flags & ^fmtPGCatalogCasts
+		ctx.flags = ctx.flags | FmtBareStrings
 		d.pgwireFormat(ctx)
 		return
 	}

